pkg/storkctl/action.go

```diff
@@ -20,99 +20,144 @@ const (
+	performFailoverCommand.Flags().BoolVar(&skipDeactivateSource, "skip-deactivate-source", false, "If present, applications in the source cluster will not be scaled down as part of the failover.")
+	performFailoverCommand.Flags().StringVarP(&referenceMigrationSchedule, "migration-reference", "m", "", "Specify the migration schedule to failover. Also specify the namespace of this migrationSchedule using the -n flag")
+	performFailoverCommand.Flags().StringSliceVar(&includeNamespaceList, "include-namespaces", nil, "Specify the comma-separated list of subset namespaces to be failed over. By default, all namespaces part of the migrationSchedule are failed over")
+	performFailoverCommand.Flags().StringSliceVar(&excludeNamespaceList, "exclude-namespaces", nil, "Specify the comma-separated list of subset namespaces to be skipped during the failover. By default, all namespaces part of the migrationSchedule are failed over")
+	return performFailoverCommand
 }
 
-func isActionIncomplete(action *storkv1.Action) bool {
-	return action.Status.Status == storkv1.ActionStatusScheduled || action.Status.Status == storkv1.ActionStatusInProgress
-}
-
-// check if there is already an Action scheduled or in-progress
-func getAnyIncompleteAction(namespace string) *storkv1.Action {
-	actionList, err := storkops.Instance().ListActions(namespace)
-	if err != nil {
-		util.CheckErr(err)
-		return nil
+// given the name, namespace, actionType and actionParameters create an Action CR
+func createActionCR(actionName string, namespace string, actionType storkv1.ActionType, actionParameters storkv1.ActionParameter) (*storkv1.Action, error) {
+	action := storkv1.Action{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      actionName,
+			Namespace: namespace,
+		},
+		Spec: storkv1.ActionSpec{
+			ActionType:      actionType,
+			ActionParameter: &actionParameters,
+		},
+		Status: storkv1.ActionStatus{Status: storkv1.ActionStatusInitial},
 	}
-	for _, action := range actionList.Items {
-		if isActionIncomplete(&action) {
-			return &action
-		}
+	actionObj, err := storkops.Instance().CreateAction(&action)
+	if err != nil {
+		return nil, err
 	}
-	return nil
+	return actionObj, nil
 }
 
-func newActionName(action storkv1.ActionType) string {
-	return strings.Join([]string{string(action), time.Now().Format(nameTimeSuffixFormat)}, "-")
+func getActionStatusMessage(action *storkv1.Action) string {
+	return fmt.Sprintf("To check %v status use the command : `storkctl get %v %v -n %v`", action.Spec.ActionType, action.Spec.ActionType, action.Name, action.Namespace)
 }
 
-func getDescribeActionMessage(action *storkv1.Action) string {
-	return "To check Action status use: " +
-		fmt.Sprintf("kubectl describe action %v -n %v", action.Name, action.Namespace)
+func getActionName(actionType storkv1.ActionType, referenceResourceName string) string {
+	return strings.Join([]string{string(actionType), referenceResourceName, GetCurrentTime().Format(nameTimeSuffixFormat)}, "-")
```

Comment:
let's trim <actionType>-<referenceResourceName> so that the whole string including the time fits into the max chars. Also let's put these kind of functions in utils.

Reply:
I will trim it according to 253 chars